agent/wptdriver/web_browser.cc

```diff
@@ -232,7 +232,12 @@ bool WebBrowser::RunAndWait() {
           ret = true;
           _status.Set(_T("Waiting up to %d seconds for the test to complete"), 
                       (_test._test_timeout / SECONDS_TO_MS) * 2);
-          DWORD wait_time = _test._test_timeout * 2;
+          DWORD wait_time;
+          if (_test._overall_timeout_in_ms != 0 &&
+            _test._overall_timeout_in_ms < _test._test_timeout * 2)
+            wait_time = _test._overall_timeout_in_ms; //Strict limit on this value
+          else
+            wait_time = _test._test_timeout * 2;

```

Comment:
Why not keep using _test_timeout as the overall timeout?  *2 is probably a bit much but adding 30-60 seconds on whatever it is set to to allow for browser startup and data collection should be reasonable.